exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/StringFunctions.java

```diff
@@ -434,6 +434,64 @@ public void eval() {
 
   }
 
+  @FunctionTemplate(name = "split_part", scope = FunctionScope.SIMPLE, nulls = NullHandling.NULL_IF_NULL,
+    outputWidthCalculatorType = OutputWidthCalculatorType.CUSTOM_FIXED_WIDTH_DEFAULT)
+  public static class SplitPartStartEnd implements DrillSimpleFunc {
+    @Param
+    VarCharHolder in;
+    @Param
+    VarCharHolder delimiter;
+    @Param
+    IntHolder start;
+    @Param
+    IntHolder end;
+
+    @Workspace
+    com.google.common.base.Splitter splitter;
+
+    @Workspace
+    com.google.common.base.Joiner joiner;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Output
+    VarCharHolder out;
+
+    @Override
+    public void setup() {
+      if (start.value < 1) {
```

Comment:
What happens if the `start` and `end` values are dynamic?  I'd think you'd want this logic to be in the `eval()` function.

Reply:
Hi cgivre, thanks for your review. I don't understand your question: "What happens if the start and end values are dynamic?". In my option, SplitPartStartEnd is choosen as the function only when all four parameters are provided, so what does dynamic mean here?

Reply:
> Hi cgivre, thanks for your review. I don't understand your question: "What happens if the start and end values are dynamic?". In my option, SplitPartStartEnd is choosen as the function only when all four parameters are provided, so what does dynamic mean here?

Hi Leon, 
What I was asking here is that you do all the validation checking in the `setup()` function which is executed once.   If a user executes the following query:

```sql
SELECT split_part(<text>, <delim>, <expression>, <expression>)
```
Those expressions will only be validated on the first row.  This is fine if they are constants, but if they are say, values from other columns, or something that is computed on the fly, the I don't think the function will do what you are expecting it to do.  I'd suggest moving all that logic to the `eval()` method.

Reply:
Got it, thanks.